setup.py

```diff
@@ -8,97 +12,190 @@ def get_advanced_templates():
-            'coreos/cloud-config.yaml'
-        ] + get_advanced_templates(),
-        'pkgpanda': [
-            'docker/dcos-builder/Dockerfile'
-        ]
-    },
-    zip_safe=False
-)
+        packages=[
+            'dcos_installer',
+            'gen',
+            'gen.build_deploy',
+            'pkgpanda',
+            'pkgpanda.build',
+            'pkgpanda.http',
+            'release',
+            'release.storage',
+            'ssh'],
+        install_requires=[
+            'aiohttp==0.22.5',
+            'analytics-python',
+            'coloredlogs',
+            'Flask',
+            'flask-compress',
+            'urllib3',
+            'chardet',
+            'PyJWT',
+            # Pins taken from 'azure==2.0.0rc4'
+            'msrest==0.4.17',
+            'msrestazure==0.4.15',
+            'azure-common==1.1.4',
+            'azure-storage==0.32.0',
+            'azure-mgmt-network==0.30.0rc4',
+            'azure-mgmt-resource==0.30.0rc4',
+            'boto3',
+            'botocore',
+            'checksumdir',
+            'coloredlogs',
+            'docopt',
+            'passlib',
+            'py',
+            'pytest',
+            'pyyaml',
+            'requests==2.18.4',
+            'retrying',
+            'schema',
+            'keyring==9.1',  # FIXME: pin keyring to prevent dbus dep
+            'teamcity-messages'],
+        entry_points={
+            'console_scripts': [
+                'release=release:main',
+                'pkgpanda=pkgpanda.cli:main',
+                'mkpanda=pkgpanda.build.cli:main',
+                'dcos_installer=dcos_installer.cli:main',
+            ],
+        },
+        package_data={
+            'gen': [
+                'ip-detect/aws.sh',
+                'ip-detect/aws_public.sh',
+                'ip-detect/azure.sh',
+                'ip-detect/vagrant.sh',
+                'fault-domain-detect/cloud.sh',
+                'cloud-config.yaml',
+                'dcos-config.yaml',
+                'dcos-metadata.yaml',
+                'dcos-services.yaml',
+                'aws/dcos-config.yaml',
+                'aws/templates/aws.html',
+                'aws/templates/cloudformation.json',
+                'azure/cloud-config.yaml',
+                'azure/azuredeploy-parameters.json',
+                'azure/templates/acs.json',
+                'azure/templates/azure.html',
+                'azure/templates/azuredeploy.json',
+                'build_deploy/bash/dcos_generate_config.sh.in',
+                'build_deploy/bash/Dockerfile.in',
+                'build_deploy/bash/installer_internal_wrapper.in',
+                'build_deploy/bash/dcos-launch.spec',
+                'coreos-aws/cloud-config.yaml',
+                'coreos/cloud-config.yaml'
+            ] + get_advanced_templates(),
+            'pkgpanda': [
+                'docker/dcos-builder/Dockerfile'
+            ]
+        },
+        zip_safe=False
+    )
```

Comment:
This is duplicating way too much code. Looks like we only need to change the set of package dependencies based on the platform.

Reply:
@paulcallen -- can we simplify this?